wallet-test/src/test/scala/org/bitcoins/wallet/dlc/DLCExecutionTest.scala

```diff
@@ -0,0 +1,272 @@
+      assert(ScriptInterpreter.checkTransaction(closeTx))
+
+      val txSigComponent =
+        WitnessTxSigComponent(tx.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              fundingTx.outputs.head,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(PreExecutionScriptProgram(txSigComponent)))
+
+      val output = if (asInitiator) tx.outputs.head else tx.outputs.last
+
+      val closeSigComponent =
+        WitnessTxSigComponent(closeTx.asInstanceOf[WitnessTransaction],
+                              UInt32.zero,
+                              output,
+                              flags)
+      assert(
+        ScriptInterpreter.runVerify(
+          PreExecutionScriptProgram(closeSigComponent)))
+    }
+  }
+
+  it must "get the correct funding transaction" in { wallets =>
+    val dlcA = wallets._1.wallet
+    val dlcB = wallets._2.wallet
+
+    for {
+      offerOpt <- dlcA.dlcOfferDAO.findByEventId(eventId)
+      acceptOpt <- dlcB.dlcAcceptDAO.findByEventId(eventId)
+
+      inputsA <- dlcA.dlcInputsDAO.findByEventId(eventId)
+      inputsB <- dlcB.dlcInputsDAO.findByEventId(eventId)
+
+      fundingTx <- dlcB.getDLCFundingTx(eventId)
+    } yield {
+      assert(offerOpt.isDefined)
+      assert(acceptOpt.isDefined)
+
+      val offer = offerOpt.get
+      val accept = acceptOpt.get
+
+      val comparableInputsA = inputsA.sortBy(_.outPoint.hex)
+      val comparableInputsB =
+        inputsB
+          .sortBy(_.outPoint.hex)
+          .map(_.copy(sigs = Vector.empty)) // initiator will not have funding sigs
+
+      assert(comparableInputsA == comparableInputsB)
+
+      val fundingTxOutpoints = fundingTx.inputs.map(_.previousOutput)
+      val outpointsA = inputsA.map(_.outPoint)
+      val outpointsB = inputsB.map(_.outPoint)
+
+      assert(fundingTxOutpoints.diff(outpointsA ++ outpointsB).isEmpty)
+
+      assert(fundingTx.outputs.size == 3)
+      assert(
+        fundingTx.outputs.exists(
+          _.scriptPubKey == offer.changeAddress.scriptPubKey))
+      assert(
+        fundingTx.outputs.exists(
+          _.scriptPubKey == accept.changeAddress.scriptPubKey))
+      assert(ScriptInterpreter.checkTransaction(fundingTx))
+
+      val fundingTxPrevOutputRefs = inputsA.map(_.toOutputReference) ++ inputsB
+        .map(_.toOutputReference)
+
+      val fundingTxVerify = fundingTx.inputs.zipWithIndex.forall {
+        case (input, index) =>
+          val output = fundingTxPrevOutputRefs
+            .find(_.outPoint == input.previousOutput)
+            .get
+            .output
+          val sigComponent =
+            WitnessTxSigComponent(fundingTx.asInstanceOf[WitnessTransaction],
+                                  UInt32(index),
+                                  output,
+                                  flags)
+          ScriptInterpreter.runVerify(PreExecutionScriptProgram(sigComponent))
```

Comment:
nit: This pattern is widely used, I think it might be nice to make a def that takes a tx, index, and output and does the rest. Might even be worth adding to `ScriptInterpreter` or at the very least the top of this file